File: convert_to_php.py
import os
import re

# Use index.html as the source of truth for header and footer
with open('index.html', 'r', encoding='utf-8') as f:
    index_content = f.read()

# Extract header
header_match = re.search(r'([\s\S]*?<!--End Main Header -->)', index_content)
if not header_match:
    print("Could not find header in index.html")
    exit(1)
header_html = header_match.group(1)

# Extract footer
footer_match = re.search(r'(<footer class="footer-section">[\s\S]*)', index_content)
if not footer_match:
    print("Could not find footer in index.html")
    exit(1)
footer_html = footer_match.group(1)

# Replace .html links with .php in header and footer
def replace_html_links(html_content):
    # This regex specifically targets href="...html"
    return re.sub(r'href="([^"]+)\.html"', r'href="\1.php"', html_content)

header_php = replace_html_links(header_html)
footer_php = replace_html_links(footer_html)

# Create include/ directory if it doesn't exist
os.makedirs('include', exist_ok=True)

# Write header.php and footer.php
with open('include/header.php', 'w', encoding='utf-8') as f:
    f.write(header_php)

with open('include/footer.php', 'w', encoding='utf-8') as f:
    f.write(footer_php)

print("Created include/header.php and include/footer.php")

# Now process all .html files
html_files = [f for f in os.listdir('.') if f.endswith('.html')]

for html_file in html_files:
    with open(html_file, 'r', encoding='utf-8') as f:
        content = f.read()

    # Replace header
    content = re.sub(r'^[\s\S]*?<!--End Main Header -->', '<?php include \'include/header.php\'; ?>', content)
    
    # Replace footer
    content = re.sub(r'<footer class="footer-section">[\s\S]*$', '<?php include \'include/footer.php\'; ?>', content)
    
    # Replace links
    content = replace_html_links(content)
    
    # Write to .php
    php_file = html_file.replace('.html', '.php')
    with open(php_file, 'w', encoding='utf-8') as f:
        f.write(content)
    
    # Remove .html file
    os.remove(html_file)
    print(f"Converted {html_file} to {php_file}")

print("Done converting all pages to PHP!")

File: convert_to_php.js
const fs = require('fs');
const path = require('path');

// Use index.html as the source of truth for header and footer
const indexContent = fs.readFileSync('index.html', 'utf-8');

// Extract header
const headerMatch = indexContent.match(/([\s\S]*?<!--End Main Header -->)/);
if (!headerMatch) {
    console.error("Could not find header in index.html");
    process.exit(1);
}
const headerHtml = headerMatch[1];

// Extract footer
const footerMatch = indexContent.match(/(<footer class="footer-section">[\s\S]*)/);
if (!footerMatch) {
    console.error("Could not find footer in index.html");
    process.exit(1);
}
const footerHtml = footerMatch[1];

// Replace .html links with .php in header and footer
function replaceHtmlLinks(htmlContent) {
    return htmlContent.replace(/href="([^"]+)\.html"/g, 'href="$1.php"');
}

const headerPhp = replaceHtmlLinks(headerHtml);
const footerPhp = replaceHtmlLinks(footerHtml);

// Create include/ directory if it doesn't exist
if (!fs.existsSync('include')) {
    fs.mkdirSync('include');
}

// Write header.php and footer.php
fs.writeFileSync('include/header.php', headerPhp, 'utf-8');
fs.writeFileSync('include/footer.php', footerPhp, 'utf-8');

console.log("Created include/header.php and include/footer.php");

// Now process all .html files
const files = fs.readdirSync('.');
const htmlFiles = files.filter(f => f.endsWith('.html'));

for (const htmlFile of htmlFiles) {
    let content = fs.readFileSync(htmlFile, 'utf-8');

    // Replace header
    content = content.replace(/^[\s\S]*?<!--End Main Header -->/, "<?php include 'include/header.php'; ?>");
    
    // Replace footer
    content = content.replace(/<footer class="footer-section">[\s\S]*$/, "<?php include 'include/footer.php'; ?>");
    
    // Replace links
    content = replaceHtmlLinks(content);
    
    // Write to .php
    const phpFile = htmlFile.replace('.html', '.php');
    fs.writeFileSync(phpFile, content, 'utf-8');
    
    // Remove .html file
    fs.unlinkSync(htmlFile);
    console.log(`Converted ${htmlFile} to ${phpFile}`);
}

console.log("Done converting all pages to PHP!");
